pkg/validator/convertor.go

```diff
@@ -0,0 +1,216 @@
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validator
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/munnerz/goautoneg"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+)
+
+type mediaType struct {
+	Type, SubType string
+}
+
+var scheme = runtime.NewScheme()
+var serializers = map[mediaType]runtime.Serializer{
+	{"application", "json"}: json.NewSerializer(json.DefaultMetaFactory, scheme, scheme, false),
+	{"application", "yaml"}: json.NewYAMLSerializer(json.DefaultMetaFactory, scheme, scheme),
+}
+
+func getInputSerializer(contentType string) runtime.Serializer {
+	parts := strings.SplitN(contentType, "/", 2)
+	if len(parts) != 2 {
+		return nil
+	}
+	return serializers[mediaType{parts[0], parts[1]}]
+}
+
+func getOutputSerializer(accept string) runtime.Serializer {
+	if len(accept) == 0 {
+		return serializers[mediaType{"application", "json"}]
+	}
+
+	clauses := goautoneg.ParseAccept(accept)
+	for _, clause := range clauses {
+		for k, v := range serializers {
+			switch {
+			case clause.Type == k.Type && clause.SubType == k.SubType,
+				clause.Type == k.Type && clause.SubType == "*",
+				clause.Type == "*" && clause.SubType == "*":
+				return v
+			}
+		}
+	}
+
+	return nil
+}
+
+func convert(
+	convertRequest *v1beta1.ConversionRequest,
+) *v1beta1.ConversionResponse {
+
+	var convertedObjects []runtime.RawExtension
+	for _, obj := range convertRequest.Objects {
+		cr := unstructured.Unstructured{}
+		if err := cr.UnmarshalJSON(obj.Raw); err != nil {
+			return &v1beta1.ConversionResponse{
+				Result: metav1.Status{
+					Message: fmt.Sprintf("failed to unmarshall object (%v) with error: %v", string(obj.Raw), err),
+					Status:  metav1.StatusFailure,
+				},
+			}
+		}
+
+		fromVersion := cr.GetAPIVersion()
+		toVersion := convertRequest.DesiredAPIVersion
+		convertedObject := cr.DeepCopy()
+
+		if fromVersion == toVersion {
+			return &v1beta1.ConversionResponse{
+				Result: metav1.Status{
+					Message: fmt.Sprintf("conversion from a version to itself should not call the webhook: %s", toVersion),
+					Status:  metav1.StatusFailure,
+				},
+			}
+		}
+
+		switch fromVersion {
+		case "kubedirector.hpe.com/v1beta1":
+			switch toVersion {
+			case "kubedirector.hpe.com/v1beta2":
+				spec := convertedObject.Object["spec"]
+				delete(convertedObject.Object, "spec")
+				spec.(map[string]interface{})["defaultEventList"] = []string{"configure", "addnodes", "delnodes"}
```

Comment:
I think when going from v1beta1 to v1beta2 we don't have to do anything to the spec, since new fields added in v1beta2 are optional and when omitted have the same behavior as v1beta1.